normandy/control/static/control/js/components/action_forms/HeartbeatForm.jsx

```diff
@@ -0,0 +1,105 @@
+import React from 'react'
+import { reduxForm } from 'redux-form'
+import { _ } from 'underscore'
+import { formatLabel } from '../../utils/formHelpers.js'
+
+
+class SurveyForm extends React.Component {
+  render() {
+    const { selectedSurvey, fields, showDefaults } = this.props;
+    const surveyObject = selectedSurvey || fields.defaults;
+    let headerText = 'Default Survey Values';
+    let containerClass = 'fluid-8';
+    if (selectedSurvey) {
+      headerText = selectedSurvey.title.initialValue || 'New survey';
+      containerClass += ' active';
+    }
+
+    return (
+      <div id='survey-form' className={containerClass}>
+        { selectedSurvey &&
+          <span className="return-to-defaults" onClick={showDefaults}>
+            <i className="fa fa-long-arrow-left pre"></i> Return to defaults
+          </span>
+        }
+        <h4>{headerText}</h4>
+        {
+          Object.keys(surveyObject).map(fieldName =>
```

Comment:
I am undecided as to whether I think we should stick with this (where we specify the fields in the field list we give to Redux Form, and then pull from that list dynamically here) or go with fully explicit forms (where we specify the fields in the field list as well as having individual components here for each field). If we were fully explicit this would look something like (assuming we also add a `TextField` component):

```jsx
<TextField className="row" label="message" field={surveyObject.message} />
<TextField className="row" label="thanksMessage" field={surveyObject.thanksMessage} />
<TextField className="row" label="engagementURL" field={surveyObject.engagementURL} />
```

Either would probably be fine, so maybe this should remain unchanged.